### PR TITLE
ref: change ar allowance

### LINKED ISSUE #813: Make the subscription 10Y and not infinite
- [ ] Make the subscription 10Y ==> Feedback makes sense
- [ ] Don't make the checkbox checked by default because of this current argent deployment 👇

https://github.com/starknet-id/app.starknet.id/assets/78437165/289d7466-a597-490d-bbab-aeb14903bd0d

### PR DESCRIPTION
**Pull request by fricoben**
need #814 to be close to test and it should close #813 

## Summary by CodeRabbit

- **New Features**
  - Enhanced discount and renewal components to include price validation during approval processes.

- **Bug Fixes**
  - Corrected the approval logic in various discount and renewal components to ensure accurate price handling.

- **Refactor**
  - Updated the `approve` function across multiple components to include an additional `price` parameter for better consistency and accuracy.

### CODE REVIEW

File: components/domains/steps/checkoutCard.tsx [hunk 1]
```diff
@@ -412,10 +412,14 @@ const CheckoutCard: FunctionComponent<CheckoutCardProps> = ({
     // If the user has toggled autorenewal
     if (renewalBox) {
       if (needsAllowance) {
+        const amountToApprove = String(
```
Comment (irisdv) [inline components/domains/steps/checkoutCard.tsx:415]: It approves the price of 1 year, shouldn't it be 10 years ?

**Review by irisdv — CHANGES_REQUESTED** [review]
When I test it only approves the price of 1 year, but should be 10 no ?
Also the checkbox for auto renewal is always checked by default when I test on the build.

File: components/domains/steps/checkoutCard.tsx (hunk 1 above)
Comment (fricoben) [reply components/domains/steps/checkoutCard.tsx:415]: Are you sure ? For me it works and send 10 years

**Review by irisdv — APPROVED** [review]
lgtm

**Pull request merged**